compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -610,17 +602,15 @@ object Completion:
               case name: TermName if include(denot, name) => Some((denot, name))
               case _ => None
 
-        types.flatMap { tp =>
+        types.flatMap: tp =>
           val tpe = tp.widenExpr
           tpe.membersBasedOnFlags(required = ExtensionMethod, excluded = EmptyFlags)
             .collect { case DenotWithMatchingName(denot, name) => TermRef(tpe, denot.symbol) -> name }
-        }
 
       // There are four possible ways for an extension method to be applicable
 
       // 1. The extension method is visible under a simple name, by being defined or inherited or imported in a scope enclosing the reference.
-      val termCompleter = new Completer(Mode.Term, pos, untpdPath, matches)
-      val extMethodsInScope = termCompleter.scopeCompletions.names.toList.flatMap:
```

Comment:
Actual fix, I was also thinking about adding
```scala
case d: SymDenotation if d.isTerm && d.termRef.symbol.is(Extension) && include(d, name)
``` 
but the issue was actually the whole creation of new completer